presto-main/src/main/java/com/facebook/presto/server/StatementResource.java

```diff
@@ -167,6 +167,23 @@ public Response createQuery(
                     .build());
         }
 
+        // The Teradata Presto ODBC Driver checks node version to decide the Presto's statement
+        // protocol and sends test queries about PREPARE statement.
+        // Rewrite the statement so that Presto always returns version for the compatible protocol.
+        // Ban the statement which will never be consumed by the driver.
+        if (servletRequest.getHeader("User-Agent").equals("Teradata Presto ODBC Driver")) {
```

Comment:
I think that would be a good practice in general. But for this case, I hope we can bundle all changes into a single place for easier code maintenance.